react/src/lib/components/DeckGLMap/components/Map.tsx

```diff
@@ -323,15 +323,16 @@ const Map: React.FC<MapProps> = ({
     const deckRef = useRef<DeckGLRef>(null);
     const bboxInitial: BoundingBox = [0, 0, 0, 1, 1, 1];
     const boundsInitial = bounds ?? [0, 0, 1, 1];
-
     // state for views prop of DeckGL component
     const [viewsProps, setViewsProps] = useState<ViewportType[]>([]);
     const [alteredLayers, setAlteredLayers] = useState([{}]);
 
+    const isViewsDefined = !isEmpty(views);
+
     const initialViewState = getViewState(
         boundsInitial,
-        views?.viewports[0].target,
-        views?.viewports[0].zoom,
+        isViewsDefined ? views?.viewports[0].target : undefined,
```

Comment:
Could we achieve the same result with the '?' operator, eg.

`views?.viewports?.[0].target,`

Reply:
> If I set `views` to something valid in the story, then the view does not update. Maybe there is some missing update.

just had a test on the master branch, seems like it is a long existing issue that was not caused by this change. Also find out that the zoom is being updated while the problem is about the target parameter that I put in does not update. I think we should resolve this Bug first and create another to track this view issue.
![image](https://user-images.githubusercontent.com/39239025/211017991-71a6e33e-a7e6-44ac-a5a0-bbcde5aa324c.png)